### PR TITLE
Task 05 done

### PR DESCRIPTION
**Pull request by ivanzuev78**
Задание выполнено, тест написан, вопросов нет.

### CODE REVIEW

File: HomeWork1/Tests/test_task05.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import unittest
+
+from HomeWork1.Tasks.task05 import find_maximal_subarray_sum
+
+
+class TestTask03(unittest.TestCase):
+    def setUp(self) -> None:
+
+        self.example_array = [1, 3, -1, -3, 5, 3, 6, 7]
```
Comment (Cjkjvfnby) [inline HomeWork1/Tests/test_task05.py:9]: Move this to the test.  Input can be a part of test.  Setup should be used only to share code betweeen multiple tests.

File: HomeWork1/Tests/test_task05.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    def test_example(self):
+
+        self.assertEqual(find_maximal_subarray_sum([1, 3, -1, -3, 5, 3, 6, 7], 3), 16)
```
Comment (Cjkjvfnby) [inline HomeWork1/Tests/test_task05.py:13]: I expect at least one more test. (Hint it will fail.)

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork1/Tests/test_task05.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork1/Tests/test_task05.py:9]: I had made one more test. But I don't understand which test (with the good input) will fail.

Comment (Cjkjvfnby) [reply HomeWork1/Tests/test_task05.py:9]: What about empty list?

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: HomeWork1/Tests/test_task05.py (hunk 2 above)
Comment (ivanzuev78) [reply HomeWork1/Tests/test_task05.py]: I thought about it. I didn't know how I can handle the exception. Now I guess.

File: HomeWork1/Tasks/task05.py [hunk 3]
```diff
@@ -13,4 +13,15 @@
 
 
 def find_maximal_subarray_sum(nums: List[int], k: int) -> int:
-    ...
+
+    if len(nums) == 0:
```
Comment (Cjkjvfnby) [inline HomeWork1/Tasks/task05.py:17]: `If not nums`

File: HomeWork1/Tests/test_task05.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    assert find_maximal_subarray_sum([1, 2, 3, 4], 1) == 4
+    assert find_maximal_subarray_sum([1, 2, 3, 4], 10) == 10
+
+
+def test_find_maximal_subarray_sum_empty_input():
+
+    try:
+
+        find_maximal_subarray_sum([], 10)
```
Comment (Cjkjvfnby) [inline HomeWork1/Tests/test_task05.py:19]: Is it a preferred way to check exception with unittests?

**Review by Cjkjvfnby — APPROVED** [review]
All comments are optional

**Pull request merged**